ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java

```diff
@@ -85,7 +85,8 @@
     } else {
       keyFormat = SerdeFeaturesFactory.sanitizeKeyFormat(
           this.keyFormat,
-          schema.key().size() == 1
+          schema.key().size(),
```

Comment:
Just for clarification: why do we allow change key format change in groupBy / selectKey but not in aggregate? Is the internal formats of groupBy / selectKey used for the repartition topic, not the aggregate?

Reply:
The aggregate is always preceded by a groupBy, and the key schema does not change between the groupBy and the aggregate. As long as we properly sanitize the key format in groupBy, then we do not need to re-sanitize during aggregate (the aggregate should already inherit a sanitized key format from its parent step, the groupBy).

Reply:
Yup, thanks for the clarification. I was just wondering if the key format after the "groupBy" is used in the repartition topic, or the key format after the "aggregate" is used in the repartition topic, or none were used in the repartition topic.

So it seems that still, the key format after the "aggregate" is used, but that is always the same as the one after the "groupBy", is my understanding correct?

Reply:
You're asking about the changelog topic, right? If so, you're correct that it's the key format of the "aggregate" that's used ([link](https://github.com/confluentinc/ksql/blob/b69d74cfeb3033748c11e3a4ba7924d1e4123d58/ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/StreamAggregateBuilder.java#L96)) rather than the key format of the "groupBy", and that the two key formats should always be the same as the "aggregate" inherits its formats from the "groupBy".

Reply:
Yes, I meant changelog topic :) And thanks for the pointer!